Source/WebCore/rendering/HitTestResult.cpp

```diff
@@ -542,6 +542,32 @@ void HitTestResult::enterFullscreenForVideo() const
 #endif
 }
 
+bool HitTestResult::mediaIsInVideoViewer() const
+{
+#if PLATFORM(MAC) && ENABLE(VIDEO) && ENABLE(VIDEO_PRESENTATION_MODE)
+    HTMLMediaElement* mediaElt(mediaElement());
+    return is<HTMLVideoElement>(mediaElt) && mediaElt->fullscreenMode() == HTMLMediaElementEnums::VideoFullscreenModeInWindow;
+#else
+    return false;
+#endif
+}
+
+void HitTestResult::toggleVideoViewer() const
+{
+#if PLATFORM(MAC) && ENABLE(VIDEO) && ENABLE(VIDEO_PRESENTATION_MODE)
+    auto* mediaElement(this->mediaElement());
+    auto* videoElement = dynamicDowncast<HTMLVideoElement>(*mediaElement);
```

Comment:
I think these should be RefPtr's maybe?

Reply:
I see a lot of examples in this file of raw pointers being used (for example, HitTestResult::toggleEnhancedFullscreenForVideo()). I am not sure if this is for a particular reason, or if it would still be best practice to use RefPtr's. @pxlcoder do you have thoughts?

Reply:
All new code should use `RefPtr`s. We are slowly working towards adopting smart pointers everywhere (which is why others in this file have not yet been updated).